### PR TITLE
Tests for repository passwd

### LINKED ISSUE #117: Tests for repo passwd / change password functionality

### PR DESCRIPTION
**Pull request by ma-hartma**
Closes #117

### CODE REVIEW

File: repository_test.go [hunk 1]
```diff
@@ -73,3 +73,46 @@ func TestRepositoryIsEmpty(t *testing.T) {
...
+		return
+	}
+
+	repo, err := OpenRepository(dir, testPassword)
+	if err != nil {
+		t.Errorf("Failed opening repository: %s", err)
+		return
+	}
+
+	err = repo.ChangePassword(newPassowrd)
```
Comment (penguwin) [inline repository_test.go:100]: Typo in variable name: `newPassword`

---

You could also simplify this a little bit:
```go
if !repo.ChangePassword(newPassword) {
   // ...
}
```

File: repository_test.go [hunk 2]
```diff
@@ -73,3 +73,46 @@ func TestRepositoryIsEmpty(t *testing.T) {
 		t.Error("Repository should not be empty")
 	}
 }
+
+func TestRepositoryChangePassword(t *testing.T) {
+	testPassword := "this_is_a_password"
+	newPassowrd := "this_is_another_password"
```
Comment (penguwin) [inline repository_test.go:79]: Typo in variable name: `newPassword`

File: repository_test.go [hunk 3]
```diff
@@ -73,3 +73,46 @@ func TestRepositoryIsEmpty(t *testing.T) {
...
+		return
+	}
+
+	_, err = OpenRepository(dir, testPassword)
+	if err == nil {
+		t.Errorf("Repository can still be opened with the old password after changing it: %s", err)
+		return
+	}
+
+	_, err = OpenRepository(dir, newPassowrd)
```
Comment (penguwin) [inline repository_test.go:112]: Typo in variable name: `newPassword`

**Comment by ma-hartma** [comment]
@penguwin thank your for your feedback!

**Comment by ma-hartma** [comment]
@penguwin thank you for reviewing!

File: repository_test.go [hunk 4]
```diff
@@ -73,3 +73,45 @@ func TestRepositoryIsEmpty(t *testing.T) {
...
 	}
 }
+
+func TestRepositoryChangePassword(t *testing.T) {
+	testPassword := "this_is_a_password"
+	newPassword := "this_is_another_password"
+
+	dir, err := ioutil.TempDir("", "knoxite")
+	if err != nil {
+		t.Errorf("Failed creating temporary dir for repository: %s", err)
```
Comment (muesli) [inline repository_test.go:83]: Instead of a `t.Errorf` followed by a return, I would suggest using `t.Fatalf` throughout this change.

**Pull request merged**